// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: proto/auth.proto

package auth

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on CheckTokenReq with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *CheckTokenReq) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CheckTokenReq with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in CheckTokenReqMultiError, or
// nil if none found.
func (m *CheckTokenReq) ValidateAll() error {
	return m.validate(true)
}

func (m *CheckTokenReq) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Token

	if len(errors) > 0 {
		return CheckTokenReqMultiError(errors)
	}

	return nil
}

// CheckTokenReqMultiError is an error wrapping multiple validation errors
// returned by CheckTokenReq.ValidateAll() if the designated constraints
// aren't met.
type CheckTokenReqMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CheckTokenReqMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CheckTokenReqMultiError) AllErrors() []error { return m }

// CheckTokenReqValidationError is the validation error returned by
// CheckTokenReq.Validate if the designated constraints aren't met.
type CheckTokenReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CheckTokenReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CheckTokenReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CheckTokenReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CheckTokenReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CheckTokenReqValidationError) ErrorName() string { return "CheckTokenReqValidationError" }

// Error satisfies the builtin error interface
func (e CheckTokenReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCheckTokenReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CheckTokenReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CheckTokenReqValidationError{}

// Validate checks the field values on CheckTokenRes with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *CheckTokenRes) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CheckTokenRes with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in CheckTokenResMultiError, or
// nil if none found.
func (m *CheckTokenRes) ValidateAll() error {
	return m.validate(true)
}

func (m *CheckTokenRes) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetData()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, CheckTokenResValidationError{
					field:  "Data",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, CheckTokenResValidationError{
					field:  "Data",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetData()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return CheckTokenResValidationError{
				field:  "Data",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if m.Error != nil {
		// no validation rules for Error
	}

	if len(errors) > 0 {
		return CheckTokenResMultiError(errors)
	}

	return nil
}

// CheckTokenResMultiError is an error wrapping multiple validation errors
// returned by CheckTokenRes.ValidateAll() if the designated constraints
// aren't met.
type CheckTokenResMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CheckTokenResMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CheckTokenResMultiError) AllErrors() []error { return m }

// CheckTokenResValidationError is the validation error returned by
// CheckTokenRes.Validate if the designated constraints aren't met.
type CheckTokenResValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CheckTokenResValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CheckTokenResValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CheckTokenResValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CheckTokenResValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CheckTokenResValidationError) ErrorName() string { return "CheckTokenResValidationError" }

// Error satisfies the builtin error interface
func (e CheckTokenResValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCheckTokenRes.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CheckTokenResValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CheckTokenResValidationError{}

// Validate checks the field values on CheckTokenResData with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *CheckTokenResData) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CheckTokenResData with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// CheckTokenResDataMultiError, or nil if none found.
func (m *CheckTokenResData) ValidateAll() error {
	return m.validate(true)
}

func (m *CheckTokenResData) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Account

	if len(errors) > 0 {
		return CheckTokenResDataMultiError(errors)
	}

	return nil
}

// CheckTokenResDataMultiError is an error wrapping multiple validation errors
// returned by CheckTokenResData.ValidateAll() if the designated constraints
// aren't met.
type CheckTokenResDataMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CheckTokenResDataMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CheckTokenResDataMultiError) AllErrors() []error { return m }

// CheckTokenResDataValidationError is the validation error returned by
// CheckTokenResData.Validate if the designated constraints aren't met.
type CheckTokenResDataValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CheckTokenResDataValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CheckTokenResDataValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CheckTokenResDataValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CheckTokenResDataValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CheckTokenResDataValidationError) ErrorName() string {
	return "CheckTokenResDataValidationError"
}

// Error satisfies the builtin error interface
func (e CheckTokenResDataValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCheckTokenResData.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CheckTokenResDataValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CheckTokenResDataValidationError{}
